avm-transpiler/src/utils.rs

```diff
@@ -1,33 +1,46 @@
+use acvm::acir::brillig::Opcode as BrilligOpcode;
 use log::debug;
 
-use acvm::acir::circuit::brillig::Brillig;
-use acvm::acir::circuit::Opcode;
+use acvm::acir::circuit::{Opcode, Program};
 
 use crate::instructions::AvmInstruction;
 
-/// Extract the Brillig program from its ACIR wrapper instruction.
-/// An Noir unconstrained function compiles to one ACIR instruction
-/// wrapping a Brillig program. This function just extracts that Brillig
-/// assuming the 0th ACIR opcode is the wrapper.
-pub fn extract_brillig_from_acir(opcodes: &Vec<Opcode>) -> &Brillig {
-    if opcodes.len() != 1 {
-        panic!("An AVM program should be contained entirely in only a single ACIR opcode flagged as 'Brillig'");
-    }
-    let opcode = &opcodes[0];
-    match opcode {
-        Opcode::Brillig(brillig) => brillig,
+/// Extract the Brillig program from its `Program` wrapper.
+/// Noir entry point unconstrained functions are compiled to their own list contained
+/// as part of a full program. Function calls are then accessed through a function
+/// pointer opcode in ACIR that fetches those unconstrained functions from the main list.
+/// This function just extracts Brillig bytecode, with the assumption that the
+/// 0th unconstrained function in the full `Program` structure.
+pub fn extract_brillig_from_acir_program(program: &Program) -> &[BrilligOpcode] {
+    assert_eq!(
+        program.functions.len(),
+        1,
+        "An AVM program should have only a single ACIR function flagged as 'BrilligCall'"
+    );
+    let opcodes = &program.functions[0].opcodes;
+    assert_eq!(
+        opcodes.len(),
+        1,
+        "An AVM program should have only a single ACIR function flagged as 'BrilligCall'"
+    );
+    match opcodes[0] {
+        Opcode::BrilligCall { .. } => {}
         _ => panic!("Tried to extract a Brillig program from its ACIR wrapper opcode, but the opcode doesn't contain Brillig!"),
     }
+    assert_eq!(
+        program.unconstrained_functions.len(),
+        1,
+        "An AVM program should be contained entirely in only a single `Brillig` function"
+    );
+    &program.unconstrained_functions[0].bytecode
 }
 
 /// Print inputs, outputs, and instructions in a Brillig program
-pub fn dbg_print_brillig_program(brillig: &Brillig) {
+pub fn dbg_print_brillig_program(brillig_bytecode: &[BrilligOpcode]) {
```

Comment:
I figuredi t would be fine to just print the bytecode as the AVM transpiler doesn't look to touch the inputs/outputs at all, we only print instructions for the AVM program, and you can get the Brillig inputs/outputs by printing the ACIR.